### ISSUE TITLE
must taint keyvault to change sku

### ISSUE THREAD

**Issue by ctooley21** [terraform-providers/terraform-provider-azurerm#4019]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)
Terraform v0.11.14
+ provider.azurerm v1.32.0
+ provider.external v1.2.0
+ provider.null v2.1.2
+ provider.random v2.1.2

* `azurerm_keyvault`

### Terraform Configuration Files

```hcl
resource "azurerm_key_vault" "key_vlt" {
  name                        = "GAV-ARV-${var.env}-SEC-KVT-01"
  location                    = "${azurerm_resource_group.rg_kvt.location}"
  resource_group_name         = "${azurerm_resource_group.rg_kvt.name}"
  enabled_for_disk_encryption = true
  tenant_id                   = "${data.azurerm_client_config.current.tenant_id}"  

  sku_name = "premium"
  
  tags = {
            uai = "${var.uai}"
            env = "${var.env}"
            SupportEmail = "${var.supportEmail}"
        }
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

Terraform should change the keyvault sku when we change the sku in our configuration files. Terraform tries to modify the keyvault, but the modification never works. You have to taint it for the sku to finally change

### Actual Behavior

Nothing changed

### Steps to Reproduce

1. Create keyvault with standard sku
2. Change sku to premium
3. terraform apply

### Important Factoids

N/A

### References

N/A

**Comment by sladeedmonds** [comment]
Ran in to this issue as well.  I am migrating an environment that wasn't initially deployed with Terraform.  There were some vaults that had been previously deployed with the standard sku that I first imported and then flipped to premium.  Terraform didn't actually change the sku, so I ended up changing it with azure cli.  Even after it was changed though, I continued getting output from Terraform that it was going to change it, only it wasn't reporting a change from standard to premium, but from "Premium" to "premium".  I tried updating the state file manually to reflect the premium sku in all lower-case, but Terraform apply would change it back to "Premium" in the state file and then a terraform plan would once again report that it wanted to change the sku from "Premium" to "premium".  Also, since Terraform hasn't yet deprecated sku, I did change both the sku and sku_name properties in the state file when I set premium to all lowercase.

The work-around for me in this case was to switch from sku_name to sku in the tf config, re-apply, then switch back to the sku_name syntax.  I now see in the Terraform state file all lowercase "premium" for both the sku_name and sku properties and terraform plans/applies as expected.
